src/sentry/shared_integrations/client/base.py

```diff
@@ -78,6 +85,34 @@ def finalize_request(self, prepared_request: PreparedRequest) -> PreparedRequest
         """
         return prepared_request
 
+    def _get_redis_key(self):
+        """
+        Returns the redis key for the integration or empty str if cannot make key
+        """
+        if not hasattr(self, "integration_id"):
+            return ""
+        if not self.integration_id:
+            return ""
+        return f"sentry-integration-error:{self.integration_id}"
+
+    def is_considered_error(self, e: Exception) -> bool:
+        return True
```

Comment:
Should this default to False and then be implemented by each integration?

Reply:
So far it's only called in the base client when we raise an error, not specific to an integration like ConnectionError, Timeout etc. but this can be changed if we raise errors in integrations's clients